Sources/SwifterSwift/SwiftStdlib/StringProtocolExtensions.swift

```diff
@@ -19,23 +19,11 @@ public extension StringProtocol {
     ///     - Parameter options: Options for the comparison.
     /// - Returns: The longest common suffix of the receiver and the given String
     func commonSuffix<T: StringProtocol>(with aString: T, options: String.CompareOptions = []) -> String {
-        guard !isEmpty && !aString.isEmpty else { return "" }
-
-        var idx = endIndex
-        var strIdx = aString.endIndex
-
-        repeat {
-            formIndex(before: &idx)
-            aString.formIndex(before: &strIdx)
-
-            guard String(self[idx]).compare(String(aString[strIdx]), options: options) == .orderedSame else {
-                formIndex(after: &idx)
-                break
-            }
-
-        } while idx > startIndex && strIdx > aString.startIndex
-
-        return String(self[idx...])
+        return String(zip(reversed(), aString.reversed())
+            .lazy
+            .prefix(while: { (lhs: Character, rhs: Character) in String(lhs).compare(String(rhs), options: options) == .orderedSame })
+            .map { (lhs: Character, _: Character) in lhs }
```

Comment:
Again, compiler optimisation. Saving the compiler from having to infer the type here saves a few hundred ms.

Reply:
Hum, I thought that will be the case if we revome the type annotation from the `prefix` but here the type can't be anything else since it was specified in the `prefix` closure, but it is ok was more a nit :)